From 33470177d17d54cdcb546d13a4518519e01a085d Mon Sep 17 00:00:00 2001
From: Todd Poynor <toddpoynor@google.com>
Date: Tue, 10 Jul 2012 17:58:29 -0700
Subject: [PATCH 524/944] power: smb347 Handle charger source connect events

Undervoltage IRQs are signalled only on charger disconnect, not
connect.

* Update the comments to reflect this fact.
* Check for changed connected/disconnected status on any interrupt.

Change-Id: I4d542ae5fc94ce9386fdedc298dbc7d84a4e0d6b
Signed-off-by: Todd Poynor <toddpoynor@google.com>
---
 drivers/power/smb347-charger.c | 15 ++++++++-------
 1 file changed, 8 insertions(+), 7 deletions(-)

diff --git a/drivers/power/smb347-charger.c b/drivers/power/smb347-charger.c
index 33784e5..70aecb7 100644
--- a/drivers/power/smb347-charger.c
+++ b/drivers/power/smb347-charger.c
@@ -814,14 +814,15 @@ static irqreturn_t smb347_interrupt(int irq, void *data)
 
 	/*
 	 * If we got an under voltage interrupt it means that AC/USB input
-	 * was connected or disconnected.
+	 * was disconnected.
 	 */
-	if (irqstat_e & (IRQSTAT_E_USBIN_UV_IRQ | IRQSTAT_E_DCIN_UV_IRQ)) {
-		if (smb347_update_status(smb) > 0) {
-			smb347_update_online(smb);
-			power_supply_changed(&smb->mains);
-			power_supply_changed(&smb->usb);
-		}
+	if (irqstat_e & (IRQSTAT_E_USBIN_UV_IRQ | IRQSTAT_E_DCIN_UV_IRQ))
+		ret = IRQ_HANDLED;
+
+	if (smb347_update_status(smb) > 0) {
+		smb347_update_online(smb);
+		power_supply_changed(&smb->mains);
+		power_supply_changed(&smb->usb);
 		ret = IRQ_HANDLED;
 	}
 
-- 
1.8.0

